core/src/subgraph/context/instance.rs

```diff
@@ -59,7 +63,7 @@ where
         // Create a new runtime host for each data source in the subgraph manifest;
         // we use the same order here as in the subgraph manifest to make the
         // event processing behavior predictable
-        for ds in manifest.data_sources {
+        for ds in data_sources {
```

Comment:
If manifest already has all the necessary information why pass these separate?

Reply:
What manifest has is the static data sources defined in the manifest, earlier the manifest was mutated to add in created dynamic datasources, this is no longer the case, the `data_sources` vec now has the complete static_data_source + dynamic_data_sources in it, and `manifest.data_sources` only contains the static data sources

Reply:
We are still passing the templates around, so I'm not sure about this code, if we don't need the entire manifest anymore then could we just pass the bits that are needed?

Reply:
The reason for passing `datasources` separately from the manifest was to avoid mutating the manifest to add in the dynamic data sources (which was also a TODO in a comment)

> We are still passing the templates around, so I'm not sure about this code
Templates are passed on to `SubgraphInstance` which is used in `HostBuilder`

The manifest is still used and is needed for passing down `manifest.data_sources` and `manifest.templates` to the Instance.